### PR TITLE
Simplified code

### PR DESCRIPTION
**Pull request by SeanFeldman**
- Replacing Task.Factory with Task.Run (advised)
- Removing task unwrapping
- Removing delay that doesn't seem to do anything

### CODE REVIEW

**Comment by SeanFeldman** [comment]
@serkantkaraca please review

File: test/Microsoft.Azure.EventHubs.UnitTests/EventHubClientTests.cs [hunk 1]
```diff
@@ -658,56 +658,54 @@ void ValidateRetryPolicyCustom()
...
                     var ehClient = EventHubClient.CreateFromConnectionString(this.EventHubConnectionString);
                     await ehClient.SendAsync(new EventData(Encoding.UTF8.GetBytes("Hello EventHub!")));
                 });
 
-                tasks.Add(task.Unwrap());
+                tasks.Add(task);
             }
 
-            // Let all tasks make the call around the same time.
-            await Task.Delay(10000);
```
Comment (serkantkaraca) [inline test/Microsoft.Azure.EventHubs.UnitTests/EventHubClientTests.cs]: > await Task.Delay(10000); [](start = 11, length = 25)

Please keep this as the buffer for thread initialization.

File: test/Microsoft.Azure.EventHubs.UnitTests/EventHubClientTests.cs [hunk 2]
```diff
@@ -658,56 +658,54 @@ void ValidateRetryPolicyCustom()
...
                     var ehClient = EventHubClient.CreateFromConnectionString(this.EventHubConnectionString);
                     await ehClient.GetRuntimeInformationAsync();
                 });
 
-                tasks.Add(task.Unwrap());
+                tasks.Add(task);
             }
 
-            // Let all tasks make the call around the same time.
-            await Task.Delay(10000);
```
Comment (serkantkaraca) [inline test/Microsoft.Azure.EventHubs.UnitTests/EventHubClientTests.cs]: >await Task.Delay(10000); [](start = 12, length = 24)

Please keep this as the buffer for thread initialization.

File: test/Microsoft.Azure.EventHubs.UnitTests/EventHubClientTests.cs (hunk 1 above)
Comment (SeanFeldman) [inline test/Microsoft.Azure.EventHubs.UnitTests/EventHubClientTests.cs]: reverted

File: test/Microsoft.Azure.EventHubs.UnitTests/EventHubClientTests.cs (hunk 2 above)
Comment (SeanFeldman) [inline test/Microsoft.Azure.EventHubs.UnitTests/EventHubClientTests.cs]: reverted

**Comment by serkantkaraca** [comment]
:shipit:

**Comment by SeanFeldman** [comment]
🎉 thank you :)

**Pull request merged**